x-pack/plugin/transform/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/transform/integration/TransformUpdateIT.java

```diff
@@ -0,0 +1,255 @@
+        );
+        String config = "{ \"dest\": {\"index\":\""
+            + transformDest
+            + "\"},"
+            + " \"source\": {\"index\":\""
+            + REVIEWS_INDEX_NAME
+            + "\"},"
+            + " \"pivot\": {"
+            + "   \"group_by\": {"
+            + "     \"reviewer\": {"
+            + "       \"terms\": {"
+            + "         \"field\": \"user_id\""
+            + " } } },"
+            + "   \"aggregations\": {"
+            + "     \"avg_rating\": {"
+            + "       \"avg\": {"
+            + "         \"field\": \"stars\""
+            + " } } },"
+            + "    \"max_page_search_size\": 555"
+            + " }"
+            + "}";
+
+        createTransformRequest.setJsonEntity(config);
+        Map<String, Object> createTransformResponse = entityAsMap(client().performRequest(createTransformRequest));
+
+        assertThat(createTransformResponse.get("acknowledged"), equalTo(Boolean.TRUE));
+        Request getRequest = createRequestWithAuth("GET", getTransformEndpoint() + transformId, BASIC_AUTH_VALUE_TRANSFORM_USER);
+        Map<String, Object> transforms = entityAsMap(client().performRequest(getRequest));
+        assertEquals(1, XContentMapValues.extractValue("count", transforms));
+        Map<String, Object> transform = ((List<Map<String, Object>>) XContentMapValues.extractValue("transforms", transforms)).get(0);
+        assertThat(XContentMapValues.extractValue("pivot.max_page_search_size", transform), equalTo(555));
+
+        final Request updateRequest = createRequestWithAuth(
+            "POST",
+            getTransformEndpoint() + transformId + "/_update",
+            BASIC_AUTH_VALUE_TRANSFORM_ADMIN_1
+        );
+        updateRequest.setJsonEntity("{}");
+
+        Map<String, Object> updateResponse = entityAsMap(client().performRequest(updateRequest));
+
+        assertNull(XContentMapValues.extractValue("pivot.max_page_search_size", updateResponse));
+        assertThat(XContentMapValues.extractValue("settings.max_page_search_size", updateResponse), equalTo(555));
+
+        getRequest = createRequestWithAuth("GET", getTransformEndpoint() + transformId, BASIC_AUTH_VALUE_TRANSFORM_USER);
+        transforms = entityAsMap(client().performRequest(getRequest));
+        assertEquals(1, XContentMapValues.extractValue("count", transforms));
+        transform = ((List<Map<String, Object>>) XContentMapValues.extractValue("transforms", transforms)).get(0);
+
+        assertNull(XContentMapValues.extractValue("pivot.max_page_search_size", transform));
+        assertThat(XContentMapValues.extractValue("settings.max_page_search_size", transform), equalTo(555));
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testUpdateTransferRights() throws Exception {
+        String transformId = "transform1";
+        // Note: Due to a bug the transform does not fail to start after deleting the user and therefore invalidating
```

Comment:
The transform ends up getting the set of `roles` the user has at the time of creating. It really should only fail if the roles change or are deleted. Not the user. 

This is how everything else in the stack works to my knowledge.